x/stake/keeper/validator.go

```diff
@@ -683,6 +679,8 @@ func (k Keeper) bondValidator(ctx sdk.Context, validator types.Validator) types.
 		panic(fmt.Sprintf("should not already be bonded, validator: %v\n", validator))
 	}
 
+	validator.BondHeight = ctx.BlockHeight()
```

Comment:
I'm not sure this is right. The original intent was to copy the `bondHeight` from the old validator in `bondIncrement` if already bonded, else set it to the current block height. Can we figure out why that isn't happening first?

Reply:
@cwgoes the bond height never gets set for a validator that enters the set (temporarily) as a result of another leaving...at least in the case I saw.

Reply:
That would be the bug then, makes sense.

We just need to ensure that this doesn't break the existing logic - which it might not, but it does overlap.